### ISSUE TITLE
invalid memory address or nil pointer dereference on `v0.2.0`

### ISSUE THREAD

**Issue by searsaw** [tektoncd/triggers#355]
# Expected Behavior
A PipelineRun should be created when sending a webhook from Github.

# Actual Behavior
The event listener errors out with an `invalid memory address or nil pointer dereference` panic.

# Steps to Reproduce the Problem

1. Create an event listener with a Github interceptor using `v0.2.0` syntax.
2. Send the webhook.
3. Check the logs for the error.

# Additional Info
I'm using similar markup to what's in the example in this repo for the `TriggerBinding`. The listener is pretty simple as well.

```yaml
apiVersion: tekton.dev/v1alpha1
kind: TriggerTemplate
metadata:
  name: pipeline-trigger
spec:
  params:
  - name: gitrevision
    description: The git revision
  - name: gitrepositoryurl
    description: The git repository url
  resourcetemplates:
  - apiVersion: tekton.dev/v1alpha1
    kind: PipelineRun
    metadata:
      generateName: pipeline-run-
    spec:
      serviceAccountName: service-ci
      pipelineRef:
        name: plugin-pipeline
      params:
      - name: sourceLocation
        value: location
      resources:
      - name: workspace
        resourceSpec:
          type: git
          params:
          - name: url
            value: $(params.gitrepositoryurl)
          - name: revision
            value: $(params.gitrevision)
      - name: dockerImage
        resourceSpec:
          type: image
          params:
          - name: url
            value: my-awesome-image
---
apiVersion: tekton.dev/v1alpha1
kind: TriggerBinding
metadata:
  name: pipeline-trigger
spec:
  params:
  - name: gitrevision
    value: $(body.head_commit.id)
  - name: gitrepositoryurl
    value: $(body.repository.url)
---
apiVersion: tekton.dev/v1alpha1
kind: EventListener
metadata:
  name: pipeline-trigger
spec:
  serviceAccountName: trigger-listener
  triggers:
  - name: github-repo-information
    interceptors:
    - github:
        secretRef:
          secretName: github-shared-secret
          secretKey: shared
          namespace: tekton-pipelines
        eventTypes:
        - push
    bindings:
    - name: pipeline-trigger
    template:
      name: pipeline-trigger
```

Here are the logs I'm getting. It appears to be an issue with the request being passed down to the interceptor.

```
{"level":"info","caller":"logging/config.go:100","msg":"Successfully created the logger.","knative.dev/jsonconfig":"{\"level\": \"info\",\"development\": false,\"sampling\": {\"initial\": 100,\"thereafter\": 100},\"outputPaths\": [\"stdout\"],\"errorOutputPaths\": [\"stderr\"],\"encoding\": \"json\",\"encoderConfig\": {\"timeKey\": \"\",\"levelKey\": \"level\",\"nameKey\": \"logger\",\"callerKey\": \"caller\",\"messageKey\": \"msg\",\"stacktraceKey\": \"stacktrace\",\"lineEnding\": \"\",\"levelEncoder\": \"\",\"timeEncoder\": \"\",\"durationEncoder\": \"\",\"callerEncoder\": \"\"}}"}
{"level":"info","caller":"logging/config.go:101","msg":"Logging level set to info"}
{"level":"info","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: open /var/run/ko/HEAD: no such file or directory"}
{"level":"info","logger":"eventlistener","caller":"logging/logging.go:46","msg":"Starting the Configuration eventlistener","knative.dev/controller":"eventlistener"}
{"level":"info","logger":"eventlistener","caller":"eventlistenersink/main.go:71","msg":"EventListener pod started","knative.dev/controller":"eventlistener"}
{"level":"info","logger":"eventlistener","caller":"eventlistenersink/main.go:97","msg":"Listen and serve on port 8080","knative.dev/controller":"eventlistener"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x128e438]

goroutine 35 [running]:
github.com/tektoncd/triggers/pkg/interceptors/github.(*Interceptor).ExecuteTrigger(0xc00031e360, 0xc00047a300, 0x0, 0x0, 0x0)
	/workspace/go/src/github.com/tektoncd/triggers/pkg/interceptors/github/github.go:56 +0x3c8
github.com/tektoncd/triggers/pkg/sink.Sink.executeInterceptors(0x18685c0, 0xc00009e120, 0x180dc20, 0xc0005a4160, 0x7f90477886b0, 0xc0005857e0, 0x17eb0e0, 0xc00022a2b8, 0x1824720, 0xc0005859c0, ...)
	/workspace/go/src/github.com/tektoncd/triggers/pkg/sink/sink.go:179 +0x171
github.com/tektoncd/triggers/pkg/sink.Sink.processTrigger(0x18685c0, 0xc00009e120, 0x180dc20, 0xc0005a4160, 0x7f90477886b0, 0xc0005857e0, 0x17eb0e0, 0xc00022a2b8, 0x1824720, 0xc0005859c0, ...)
	/workspace/go/src/github.com/tektoncd/triggers/pkg/sink/sink.go:129 +0x1f9
github.com/tektoncd/triggers/pkg/sink.Sink.HandleEvent.func1(0x18685c0, 0xc00009e120, 0x180dc20, 0xc0005a4160, 0x7f90477886b0, 0xc0005857e0, 0x17eb0e0, 0xc00022a2b8, 0x1824720, 0xc0005859c0, ...)
	/workspace/go/src/github.com/tektoncd/triggers/pkg/sink/sink.go:93 +0xd2
created by github.com/tektoncd/triggers/pkg/sink.Sink.HandleEvent
	/workspace/go/src/github.com/tektoncd/triggers/pkg/sink/sink.go:92 +0x79d
```

**Comment by dibyom** [comment]
Ok, I can reproduce this :cry:

**Comment by dibyom** [comment]
Have a fix in #357 The core issue was that we were passing a request object with a drained body to the first interceptor always. In addition, were using the `GetBody` function which can sometimes be nil. 

I'll add some tests and then we'll do a `0.2.1` release. Monday is a holiday so targeting Tuesday.
